components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-node/reporting-node.component.ts

```diff
@@ -46,19 +51,35 @@ export class ReportingNodeComponent implements OnInit, OnDestroy {
     const reportQuery = this.reportQueryService.getReportQuery();
     reportQuery.filters = reportQuery.filters.concat([{type: {name: 'node_id'}, value: {id}}]);
 
-    this.statsService.getReports(reportQuery, {sort: 'latest_report.end_time', order: 'DESC'})
-    .pipe(takeUntil(this.isDestroyed))
-    .subscribe(reports => {
-      this.reports = reports;
-      const queryForReport = this.reportQueryService.getReportQueryForReport(reports[0]);
-      this.statsService.getSingleReport(reports[0].id, queryForReport)
-        .pipe(takeUntil(this.isDestroyed))
-        .subscribe(data => {
-          this.reportLoading = false;
-          this.layoutFacade.ShowPageLoading(false);
-          this.activeReport = Object.assign(reports[0], data);
-        });
+    this.page$.subscribe(page => {
```

Comment:
I cannot quite get my head around this triply-nested subscription(!), but I routinely point out in code reviews that even doubly-nested is asking for trouble because subscriptions are not actionable code per se. Rather they are setting up future actions. It is almost always possible to unnest them (by adding appropriate filters) to much more easily confirm that you are getting what you expect (not to mention making it easier to test). The only time it is not a red flag to me is if the inner subscription includes a `first()` call to make it a one-shot.
(FYI I experimented some time ago with a diagnostic tool called rxjs-spy that can help trace what actually occurs if you are interested.)

Also, note that the outermost subscription here lacks a `takeUntil(this.isDestroyed)`.

Reply:
I agree. I was worried about the nesting also but I did not know away around it because the two inner subscribes are requesting data from the backend. Where the last request is based on the first. 
Thanks for the tip on using the `first()`. I pulled out both requests into their own functions where they use `first()`. This did not eliminate the triply-nested subscriptions. It only made it more readable. 

I also added the `takeUntil(this.isDestroyed)` for the outermost subscription. Thanks for pointing that out. 

This should all be placed in NGRX, but that is beyond this task.